src/Sylius/Bundle/ApiBundle/CommandHandler/RegisterShopUserHandler.php

```diff
@@ -66,18 +80,30 @@ public function __invoke(RegisterShopUser $command): void
         $customer->setPhoneNumber($command->phoneNumber);
         $customer->setUser($user);
 
-        $this->handleVerification($user);
-
-        $this->shopUserManager->persist($user);
-    }
-
-    private function handleVerification(ShopUserInterface $user): void
-    {
         /** @var ChannelInterface $channel */
-        $channel = $this->channelContext->getChannel();
+        $channel = $this->channelRepository->findOneByCode($command->channelCode);
 
-        if (!$channel->isAccountVerificationRequired()) {
+        $this->shopUserManager->persist($user);
+        $this->shopUserManager->flush();
```

Comment:
```suggestion
```

Reply:
It is needed, otherwise if not flushed here, the Repositories in Send....EmailHandlers can't see the added ShopUser

Reply:
Have you checked again after adding the stamp?

Reply:
The flush should happen out-of-the-box after finishing handler logic

Reply:
changed, lets 🙏  to build gods 😄